### ISSUE TITLE
permission denied?

### ISSUE THREAD

**Issue by FelixPQ** [NapoleonWils0n/kodi-playercorefactory#80]
I've installed your scripts via the windows installer and it seems the only thing I can do is save the URL to the desktop and use rip-record afterwards to download the file but this is very time consuming and I lose the title of the movie. Any suggestions on how to fix this?

I'm an old guy and I just can't read what is shown, it's just to fast for me. Maybe it would be nice to have a debug mode or somekind of log. I'm sure, it's nothing and relatively easy to fix but I just can't figure it out.
Just in case, here is what I'd like to do, I'd like to be able to "extract" the urls of many films or whatever and then start (robustly) downloading all of them as efficiently as possible using one or more computers. All this in an as automated fashon as possible. Any hints on this would be appreciated, thanks

**Comment by NapoleonWils0n** [comment]
hi Felix

## permission denied

i wrote the bash scripts but another user wrote the windows installer
there is a tmp folder inside the rip-record folder that doesnt have write permissions set properly

that bug has been fixed in the latest windows installer i believe

if you raise an issue on the windows installer github page 
and explain you are getting a permission denied error when you try and record from within kodi

but that you can record on the command line ok

the guy who wrote the windows installer was a windows software engineer 
and will be able to explain how to fix the permissions error

## menu

while the video is playing in kodi if you press y on the keyboard
you should get a menu which will let you save the url, record, and record with a duration

## video titles

there is no way to get the videos title from the url unfortunately 
the title isnt stored in the url 

you can either record from within kodi or save the url and record on the command line

you can save the url from kodi and then rename the text file with movies title

## batch recording

there is script called rip-record-batch which will prompt you for a folder with text files with the urls you saved from kodi
have a look at video number 46 which shows you how to use the rip-record-batch script

the video will be saved with the date and time so they have unique file names

but there isnt a way to get the movie title from the url because it is very rarely saved in the url, 
people dont add the movie title to the url because then it usualy gets taken down for copyright infringment

## changing recording destintaion

you can change the recording destination by using a config file which is in the git repo
have a look at the ripconfig folder in the git repo and watch video 122 on how to set it up

## extracting multiple urls with wireshark

you can also use wireshark and windump to capture multiple urls from kodi at once
i have made several videos on how to set up wireshark on windows and capture multiple urls at once

basically you run wireshark and capture traffic on your network interface
then you open different videos in kodi and it will capture all the urls

then you can export all the http links from wireshark

and the use the rip-record-batch script

## sending ffmpeg command over ssh

you can also send the ffmpeg record over ssh from within kodi or on the command line to another machine
anywhere in the world and it will record on the other machine have a look at video 100

have a look in the docs folder in the git repo and you will find lots of notes
if you look in the ssh folder you will see how to set up ssh on a server and as a client

hope that helps 

sorry i go a bit fast for you
usually a lot to go thru and if i stop ill forget what i was talking about

**Comment by FelixPQ** [comment]
Napoleon,

 

Unfortunately, this is the latest edition and though I’ve changed the permission of the TMP directory, something is still not working correctly but it doesn’t matter, I can save the URL and use rip-record directly, very slow, a real pain and very awkward but it works.

I just found out about Kodi database system and tough it as lots of data about the movies and much more, all the URLs I’ve tried so far don’t work. I have somekind of custom build I pay to get stable IPTV but for now I’m interested only in movies. Let’s say the links of the database where functional, I would like to run a query to generate the video-url-xxxx-xx-xx-xx-xx-xx.txt and each night I would download a few dozen and of course I would try to set the title correctly. I guess that’s my problem and not yours. I’ll see if my guy wants to tell where he gets the URLs maybe I’ll be all right. I can always do all this manually.

Many thanks for replying, I appreciate a lot,

Yves
